Add tankerkoenig integration

guillempages opened the pull request:
**Description:**
Add a new integration for tankerkoenig.de gas station prices.

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#28661
## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

guillempages committed:
Add tankerkoenig integration

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

guillempages committed:
Add new configuration variable "stations"

Enhance the documentation for the new optional configuration variable "stations"

guillempages commented:
@frenck I updated the PR after your review, since I added a new optional configuration option in the parent PR. (It was requested by a reviewer and it made sense to me).
I'm not sure what the process is here (this is my first integration); do you need to re-approve, or will you do that once the parent PR gets merged?

Also, will the failed "/Build Failed (pull request)" check prevent the merge? I don't know what caused that or how to fix it.

guillempages committed:
Fix typos in tankerkoenig integration

frenck commented:
The PR is fine at this point and is awaiting the parent PR to get merged. Don't worry about the failed `/` build. That is a pending change out of your control.

guillempages committed:
Change integration to platform

The integration is a platform integration instead of just a sensor now.
